.github/workflows/lint-test.yml

```diff
@@ -46,25 +46,19 @@ jobs:
     steps:
       - name: Check out repository code
         uses: actions/checkout@v4
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
         with:
+          version: "0.5.30"
+          enable-cache: true
+          cache-dependency-glob: "uv.lock"
           python-version: ${{ matrix.python-version }}
 
-      - name: Install pip dependencies
-        if: steps.pip-cache.outputs.cache-hit != 'true'
-        run: |
-          pip install psycopg2 'poetry<3.0.0'
-
-      - name: Poetry install
-        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
-        run: |
-          poetry install
-
       - name: Run pre-commit hooks
         run: |
-          poetry run pre-commit run --all-files --show-diff-on-failure
+          uv run pre-commit run --all-files --show-diff-on-failure
       - name: Run tests
         run: |
-          poetry add django@^${{ matrix.django-version }}
-          poetry run tests
+          uv pip install django@^${{ matrix.django-version }}
```

Comment:
why uv pip install instead of uv add ?

Reply:
uv add changes the pyproject

Reply:
Okay but it's what we want no ? Has we do not commit but launch it with uv we want the verison of django changing inside the uv virtual env no ?
I mean what you do is doing it too. But what the point of not changing the pyproject ?

Reply:
more overhead even though it is so fast it does not matter much

Reply:
The CI do not like it. are you sur it's working the same in pip (I mean the versionning semantic)